.github/workflows/build-and-push.yml

```diff
@@ -20,29 +21,64 @@ jobs:
       run: sudo ./.github/install-etcd.sh
     - name: Build with Maven
       run: mvn -B package --file pom.xml
+
   build:
     needs: test
     runs-on: ubuntu-latest
+
     env:
-      IMAGE_NAME: kserve/modelmesh
+      IMAGE_NAME: modelmesh
+
     steps:
-    - uses: actions/checkout@v2
-    - name: Build and push runtime image
-      run: |
-        GIT_COMMIT=$(git rev-parse HEAD)
-        BUILD_ID=$(date '+%Y%m%d')-$(git rev-parse HEAD | cut -c -5)
+      - uses: actions/checkout@v2
+
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to DockerHub
+        if: github.event_name == 'push'
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_ACCESS_TOKEN }}
+
+      - name: export version variable
+        run: |
+          GIT_COMMIT=$(git rev-parse HEAD)
+          BUILD_ID=$(date '+%Y%m%d')-$(git rev-parse HEAD | cut -c -5)
+          
+          IMAGE_ID=kserve/$IMAGE_NAME
+
+          # Change all uppercase to lowercase
+          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
+
+          # Strip git ref prefix from version
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
 
-        # Strip git ref prefix from version
-        VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          # Strip "v" prefix from tag name
+          # [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
 
-        # Use Docker `latest` tag convention
-        [ "$VERSION" == "main" ] && VERSION=latest
-        echo $VERSION
+          # Use Docker `latest` tag convention
+          [ "$VERSION" == "main" ] && VERSION=latest
 
-        docker build -t ${{ env.IMAGE_NAME }}:$VERSION \
-          --build-arg imageVersion=$VERSION \
-          --build-arg buildId=${BUILD_ID} \
-          --build-arg commitSha=${GIT_COMMIT} .
+          echo "GIT_COMMIT=$GIT_COMMIT" >> $GITHUB_ENV
+          echo "BUILD_ID=$BUILD_ID" >> $GITHUB_ENV
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+          echo "IMAGE_ID=$IMAGE_ID" >> $GITHUB_ENV
 
-        docker login -u ${{ secrets.DOCKER_USER }} -p ${{ secrets.DOCKER_ACCESS_TOKEN }}
-        docker push ${{ env.IMAGE_NAME }}:$VERSION
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          platforms: linux/amd64,linux/arm64/v8,linux/ppc64le
```

Comment:
maybe we can add a comment here why s390x arch is not supported for GRPC

Reply:
```suggestion
          # for linux/s390x, maven errors due to missing io.grpc:protoc-gen-grpc-java:exe:linux-s390_64:1.51.1
          platforms: linux/amd64,linux/arm64/v8,linux/ppc64le
```